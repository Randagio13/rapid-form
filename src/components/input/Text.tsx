import { Themes, validationMethod } from 'helpers'
import { Map } from 'immutable'
import * as PropTypes from 'prop-types'
import * as React from 'react'
import { callbackOnClick, setCheckError } from 'types'

export interface ITextProps {
  children?: any,
  className?: string,
  key?: string,
  name?: string,
  type: string,
  value?: string,
  theme?: string,
  onClick?: callbackOnClick,
  setCheckError?: setCheckError
}

class Text extends React.Component<ITextProps, any> {
  static propTypes = {
    className: PropTypes.string,
    key: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number
    ]),
    name: PropTypes.string,
    setCheckError: PropTypes.func,
    style: PropTypes.object,
    theme: PropTypes.string,
    type: PropTypes.oneOf([
      'text', 'file', 'password'
    ]).isRequired
  }
  public render (): JSX.Element {
    return this.handleRenderByTheme()
  }
  private handleValidation = (event: any): any => {
    const { setCheckError: checkError, ...props } = this.props
    const method = Reflect.get(this.props, 'data-validation')
    const val = Reflect.get(event.target, 'value')
    const required = Reflect.get(props, 'required')
    const key = Reflect.get(props, 'data-key')
    const isValid = validationMethod(method, val)
    if (!method || !required) {
      return checkError({...props, value: val}, key, Map())
    }
    checkError({...props, value: val}, key, isValid)
  }
  private handleRenderByTheme = (): JSX.Element => {
    const { setCheckError: checkError, theme, type, ...allProps } = this.props
    const themeClass = new Themes(theme)
    const cmp = <input {...allProps} onChange={this.handleValidation} />
    const props = { onChange: this.handleValidation, ...allProps }
    return themeClass.renderField(type, props, cmp)
  }
}

export default Text
