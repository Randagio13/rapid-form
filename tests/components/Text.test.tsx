import { shallow } from 'enzyme'
import * as React from 'react'
import { create } from 'react-test-renderer'
import Text from '../../src/components/input/Text'

export const testOnSubmit = (event: any) => {
  alert(event)
}

const component = create(
  <Text name='text' type='text' />
).toJSON()

test('Snapshot', () => {
  expect(component).toMatchSnapshot()
})

test('Check property component', () => {
  expect(component.type).toEqual('input')
  expect(component.props.type).toEqual('text')
  expect(component.props.name).toEqual('text')
})
