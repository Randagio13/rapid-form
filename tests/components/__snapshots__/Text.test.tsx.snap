// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<input
  name="text"
  type="text"
/>
`;
